packages/flutter_tools/lib/src/version.dart

```diff
@@ -145,30 +145,21 @@ class FlutterVersion {
   static FlutterVersion get instance => context.putIfAbsent(FlutterVersion, () => new FlutterVersion(const Clock()));
 
   /// Return a short string for the version (`alpha/a76bc8e22b`).
-  static String getVersionString({ bool whitelistBranchName: false }) {
-    String commit = _shortGitRevision(_runSync(<String>['git', 'rev-parse', 'HEAD']));
-    commit = commit.isEmpty ? 'unknown' : commit;
-
-    final String branch = getBranchName(whitelistBranchName: whitelistBranchName);
-
-    return '$branch/$commit';
+  String getVersionString({bool whitelistBranchName: false}) {
+    return '${getBranchName(whitelistBranchName: whitelistBranchName)}/$frameworkRevisionShort';
   }
 
   /// Return the branch name.
   ///
   /// If whitelistBranchName is true and the branch is unknown,
   /// the branch name will be returned as 'dev'.
-  static String getBranchName({ bool whitelistBranchName: false }) {
-    String branch = _runSync(<String>['git', 'rev-parse', '--abbrev-ref', 'HEAD']);
```

Comment:
This removed functionality that if the developer is on a local branch, we'll report `dev`.

Rather than remove this functionality, let's make it an instance method and thus part of the interface, then override it within Google to just return `channel`

Reply:
Yes, that's what I have done in this PR. There should be no static methods here that call 'git'.

Reply:
We will still report dev if channel is empty or it is not in the approved list of channels.

Reply:
But in the case of this method, we *were* checking the local branch name here, and now we're just using the channel, which is not always the same.  I'm saying we your change to make `getBranchName` non-static LGTM, but we should keep the old implementation that actually checked the branch name.

Reply:
Did not know about the difference of branch vs channel (comments made it look like they are the same). I now have a separate _branch var that's populated in the ctr and getBranchName() uses it. PTAL.